cub/benchmarks/bench/copy/memcpy.cu

```diff
@@ -0,0 +1,272 @@
+                                          buff_delay_constructor_t,
+                                          block_delay_constructor_t>;
+
+    using AgentLargeBufferPolicyT =
+      cub::detail::AgentBatchMemcpyLargeBuffersPolicy<TUNE_LARGE_THREADS,
+                                                      TUNE_LARGE_BUFFER_BYTES_PER_THREAD>;
+  };
+
+  using MaxPolicy = policy_t;
+};
+#endif
+
+template <class T, class OffsetT>
+void copy(nvbench::state &state, nvbench::type_list<T, OffsetT>)
+{
+  using offset_t           = OffsetT;
+  using it_t               = T *;
+  using input_buffer_it_t  = it_t *;
+  using output_buffer_it_t = it_t *;
+  using buffer_size_it_t   = offset_t *;
+  using buffer_offset_t    = std::uint32_t;
+  using block_offset_t     = std::uint32_t;
+
+  const auto elements        = static_cast<std::size_t>(state.get_int64("Elements{io}"));
+  const auto max_buffer_size = static_cast<std::size_t>(state.get_int64("MaxBufferSize"));
+  const auto min_buffer_size_ratio =
+    static_cast<std::size_t>(state.get_int64("MinBufferSizeRatio"));
+  const auto min_buffer_size =
+    static_cast<std::size_t>(static_cast<double>(max_buffer_size) / 100.0) * min_buffer_size_ratio;
+
+  constexpr bool is_memcpy = true;
+
+#if !TUNE_BASE
+  using policy_t = policy_hub_t;
+#else
+  using policy_t = cub::detail::DeviceBatchMemcpyPolicy<buffer_offset_t, block_offset_t>;
+#endif
+
+  using dispatch_t = cub::detail::DispatchBatchMemcpy<input_buffer_it_t,
+                                                      output_buffer_it_t,
+                                                      buffer_size_it_t,
+                                                      buffer_offset_t,
+                                                      block_offset_t,
+                                                      policy_t,
+                                                      is_memcpy>;
+
+  thrust::device_vector<T> input_buffer(elements);
+  thrust::device_vector<T> output_buffer(elements);
+  thrust::device_vector<offset_t> offsets =
+    gen_uniform_offsets<offset_t>(seed_t{}, elements, min_buffer_size, max_buffer_size);
+  gen(seed_t{}, input_buffer);
+
+  T *d_input_buffer   = thrust::raw_pointer_cast(input_buffer.data());
+  T *d_output_buffer  = thrust::raw_pointer_cast(output_buffer.data());
+  offset_t *d_offsets = thrust::raw_pointer_cast(offsets.data());
+
+  const auto buffers = offsets.size() - 1;
+
+  thrust::device_vector<offset_t> buffer_sizes(buffers);
+  thrust::tabulate(buffer_sizes.begin(),
+                   buffer_sizes.end(),
+                   offset_to_bytes_t<T, offset_t>{d_offsets});
+
+  thrust::device_vector<it_t> input_buffers(buffers);
+  thrust::tabulate(input_buffers.begin(),
+                   input_buffers.end(),
+                   offset_to_ptr_t<T, offset_t>{d_input_buffer, d_offsets});
+
+  thrust::device_vector<it_t> output_buffers(buffers);
+
+  if (state.get_int64("RandomizeOutput"))
```

Comment:
The existing `Memcpy::Batched` `thrust_benchmark` was having `random reads` _and_ `random writes` for the `random` configuration. Lacking usage insights, I think we should treat all workloads equally important. So, having sequential reads and random writes is just as fine for me. Just wanted to point out the difference for perf traceability.

Reply:
Decided to be on the safe side and randomize inputs as well. Thanks!